src/StudyRegistration/utils.tsx

```diff
@@ -33,26 +33,45 @@ export const preprocessStudyRegistrationMetadata = async (username, metadataID,
     }
     metadataToUpdate[STUDY_DATA_FIELD][studyRegistrationValidationField] = true;
     metadataToUpdate[STUDY_DATA_FIELD][studyRegistrationTrackingField] = username;
-    metadataToUpdate[STUDY_DATA_FIELD] = { ...metadataToUpdate[STUDY_DATA_FIELD], ...updatedValues };
+
+    // add all repository_study_ids as separate objects
+    if (updatedValues.repository_study_ids?.length > 0) {
+      const tempStudyIDObj = updatedValues.repository_study_ids.map((studyId) => ({
+        repository_name: updatedValues.repository,
+        repository_study_ID: studyId,
+      }));
+      metadataToUpdate[STUDY_DATA_FIELD].study_metadata.metadata_location.data_repositories = tempStudyIDObj;
+    }
+    metadataToUpdate[STUDY_DATA_FIELD].study_metadata.metadata_location.clinical_trials_study_ID = updatedValues.clinical_trials_id;
+    metadataToUpdate.clinicaltrials_gov = updatedValues.clinicaltrials_gov;
```

Comment:
This will make all registered metadata has a `clinicaltrials_gov` section in the metadata even though it could be empty? I think we should stick to the existing behavior where only those studies with a ctgov ID get a `clinicaltrials_gov` section in there metadata after registeration